core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -293,6 +293,24 @@ private[zio] final class FiberContext[E, A](
                   val io = curIo.asInstanceOf[ZIO.SuspendWith[Any, E, Any]]
 
                   curIo = io.f(platform)
+
+                case ZIO.Tags.FiberRefNew =>
+                  val io = curIo.asInstanceOf[ZIO.FiberRefNew[Any]]
+
+                  val fiberRef = new FiberRef[Any](io.initialValue)
+                  fiberRefLocals.put(fiberRef, (io.initialValue, fiberId))
+
+                  curIo = nextInstr(fiberRef)
+
+                case ZIO.Tags.FiberRefModify =>
+                  val io = curIo.asInstanceOf[ZIO.FiberRefModify[Any, Any]]
+
+                  val oldValue           = Option(fiberRefLocals.get(io.fiberRef))
+                  val (result, newValue) = io.f(oldValue.map(_._1).getOrElse(io.fiberRef.initial))
```

Comment:
Ah, I see why `initial` is useful inside `FiberRef`.

I guess it gives us the semantic that: if some unrelated fiber accesses a `FiberRef`, it gets the starting value, because there's no direct lineage from it to the creator of the `FiberRef`. 

That seems useful, so in hindsight, I would not delete `initial`, even though it is a potential leak.

Reply:
Yes, exactly!